torchvision/transforms/v2/functional/_geometry.py

```diff
@@ -159,21 +159,30 @@ def vertical_flip_video(video: torch.Tensor) -> torch.Tensor:
 
 
 def _compute_resized_output_size(
-    canvas_size: Tuple[int, int], size: List[int], max_size: Optional[int] = None
+    canvas_size: Tuple[int, int], size: Optional[List[int]], max_size: Optional[int] = None
 ) -> List[int]:
     if isinstance(size, int):
         size = [size]
+    elif size is None:
+        if isinstance(max_size, int):
+            h, w = canvas_size
+            short, long = (w, h) if w <= h else (h, w)
+            new_short, new_long = int(max_size * short / long), max_size
+            new_w, new_h = (new_long, new_short) if w <= h else (new_short, new_long)
+            size = [new_w, new_h]
```

Comment:
It seems that this logic should be implemented in `__compute_resized_output_size` instead, because this is where the rest of the size computation logic resides.
Since `__compute_resized_output_size` is also used for `v1` and we don't want v1 to support None, maybe we can just add a flag `allow_size_none=False`  which we set to `True` here in v2 only?

Reply:
That makes sense - I didn't want to change the v1 code originally but this solution works. I'll make the change.